### ISSUE TITLE
Disable heroCard after the first usage

### ISSUE THREAD

**Issue by Ydr** [microsoft/BotFramework-Composer#9232]
Hello team,

I am trying to use the dialogs with a heroCard but I am having a problem. When the user selects one option, PVA recognizes it, but since the adaptive dialog stays enabled, the same user can select another option, which will break the PVA logic. I have tried to add a "End Dialog" but it keeps enabled even with the action. Any ideas to overcome this behavior? Thanks in advance!

**Comment by ramfattah** [comment]
Hi @Ydr 

Could you please provide the [steps to reproduce this issue](https://github.com/microsoft/BotFramework-Composer/blob/main/.github/ISSUE_TEMPLATE/bot-framework-composer-bug.md#:~:text=To%20Reproduce,See%20error) along with your composer version?

**Comment by Ydr** [comment]
Thank you @ramfattah 

Sure! So what I am trying to do is simple, basically what's shared in this blog: https://www.bendenblanken.com/extend-power-virtual-agent-with-vertical-buttons-and-make-this-reusable/ which works like a charm.

The problem is when I want to close the dialog. My goal was to just show one time the questions and after the user clicks on one of the options, make the dialog disappear or block the action, because if it stays enabled, the user can scroll up and select other options, which are going to break the PVA logic.

This is the Framework Composer configuration:

![image](https://user-images.githubusercontent.com/1464184/174394870-36dadd41-3b01-45b1-97f5-abd83c70ca9a.png)

This is the PVA result (Which is good):

![image](https://user-images.githubusercontent.com/1464184/174395551-2c24c2e2-6ab8-4979-a4c5-8e9bfcb5673d.png)

I can select an option, the problem is that after selecting the option the Dialog continues enabled, and I can select more.

Bot Framework Composer version: 
Version:  2.1.2
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

Thank you in advance!

**Comment by Ydr** [comment]
@ramfattah thank you so much for the information. Indeed I am using the Omnichannel snippet to display the bot information https://docs.microsoft.com/en-us/dynamics365/customer-service/embed-chat-widget-portal. I believe that makes it a WebChat, right? (I do apologize since I am starting to use the technology and Composer, I am starting to get used to it).

Your confirmation may help me to find the proper way to add the code shared and try to test it (If feasible). Thanks!

**Comment by ramfattah** [comment]
@Ydr thanks, glad to help.
I'm closing this for now. Feel free to re-open if needed.

**Comment by Ydr** [comment]
@ramfattah Can you please help me to confirm my last comment? Thank you in advance!

**Comment by Ydr** [comment]
Thank you @ramfattah, It is weird because I don't know if the Omnichannel snippet is Direct Line behind the scenes (That's new for me) Omnichannel provides a snippet, that snippet is used in a webpage, like this:

![image](https://user-images.githubusercontent.com/1464184/175102092-28b39428-c1be-4751-8356-f289310332a3.png)

PVA bots are behind the scenes in the Omnichannel configuration, added to particular Work Streams. My question is where should I find the source code to add this https://github.com/microsoft/BotFramework-WebChat/blob/main/samples/05.custom-components/l.disable-adaptive-cards/index.html if we are only using the Omnichannel snippet in a Web Page (also that means that I don't know if that snippet qualifies as Web Chat)?

Thank you so much in advance!

**Comment by ramfattah** [comment]
Hi @Ydr, 

From reading the following doc:
https://docs.microsoft.com/en-us/dynamics365/customer-service/embed-chat-widget-portal

looks like if using the above omnichannel `<script` snippet, it might not work as mentioned it Embeds a chat widget in your web portal, and I don't think you can fully customize a widget (i.e. disabling card options after first use).

**Comment by Ydr** [comment]
Thank you so much @ramfattah, yes that's the problem. I don't have a full BOT created that I can customize, it is a snipped provided by Omnichannel. From Bot Framework Composer is there a way to customize the heroCard? Or an option that may allow the card to be disabled beside your suggestions, now that you know better the context where I am working?

Thank you so much for your replies!!!

**Comment by ramfattah** [comment]
Hi @Ydr, 

In regards to further customizing the the heroCard in Composer, I will discuss this internally with the team and report back. Thank you.

**Comment by Ydr** [comment]
That would be awesome @ramfattah ! The best would be that the heroCard could have a setting that after it's clicked the first time, blocks the rest of the options... I believe that will be a nice feature, that will resolve the problem with the built-in questioning provided by PVA. Thank you!! I will be pendent.

**Comment by ramfattah** [comment]
There is a possible working solution to this issue which requires using a Dynamics 365 Web Resource component instead of an iframe component. Would it be possible for you to create and utilize a [Web Resource](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/customize/create-edit-web-resources?view=op-9-1) instead?



In short, the solution requires using a custom-built HTML page using [BotFramework-WebChat](https://github.com/Microsoft/BotFramework-WebChat#bot-framework-web-chat) which would allow the ability to disable card buttons after being clicked, as per this sample, [05.custom-components/l.disable-adaptive-cards](https://github.com/microsoft/BotFramework-WebChat/tree/main/samples/05.custom-components/l.disable-adaptive-cards). Then, you just need to store the `.html` page somewhere accessible and pass that page into the iframe via the 'src' property, like:

      <iframe src=[HTML LINK TO PAGE]></frame> 


I tested this with a colleague who stored the page in Azure blob storage. We created a shared access signature for it which allowed it to display in the Dynamics 365 Web Resource. If this would be possible for you, I can get you more detailed steps for setting up.
